<center>
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/images/SN_web_lightmode.png" width="300" alt="cognitiveclass.ai logo">
</center>

<h1 align=center><font size = 5>Assignment: Notebook for Peer Assignment</font></h1>


# Introduction

Using this Python notebook you will:

1.  Understand three Chicago datasets
2.  Load the three datasets into three tables in a SQLIte database
3.  Execute SQL queries to answer assignment questions


## Understand the datasets

To complete the assignment problems in this notebook you will be using three datasets that are available on the city of Chicago's Data Portal:

1.  <a href="https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Socioeconomic Indicators in Chicago</a>
2.  <a href="https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Public Schools</a>
3.  <a href="https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01">Chicago Crime Data</a>

### 1. Socioeconomic Indicators in Chicago

This dataset contains a selection of six socioeconomic indicators of public health significance and a “hardship index,” for each Chicago community area, for the years 2008 – 2012.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2](https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 2. Chicago Public Schools

This dataset shows all school level performance data used to create CPS School Report Cards for the 2011-2012 school year. This dataset is provided by the city of Chicago's Data Portal.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t](https://data.cityofchicago.org/Education/Chicago-Public-Schools-Progress-Report-Cards-2011-/9xs2-f89t?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)

### 3. Chicago Crime Data

This dataset reflects reported incidents of crime (with the exception of murders where data exists for each victim) that occurred in the City of Chicago from 2001 to present, minus the most recent seven days.

A detailed description of this dataset and the original dataset can be obtained from the Chicago Data Portal at:
[https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2](https://data.cityofchicago.org/Public-Safety/Crimes-2001-to-present/ijzp-q8t2?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ)


### Download the datasets

This assignment requires you to have these three tables populated with a subset of the whole datasets.

In many cases the dataset to be analyzed is available as a .CSV (comma separated values) file, perhaps on the internet. Click on the links below to download and save the datasets (.CSV files):

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCensusData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Census Data</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoPublicSchools.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Public Schools</a>

*   <a href="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/FinalModule_Coursera_V5/data/ChicagoCrimeData.csv?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01" target="_blank">Chicago Crime Data</a>

**NOTE:** Ensure you have downloaded the datasets using the links above instead of directly from the Chicago Data Portal. The versions linked here are subsets of the original datasets and have some of the column names modified to be more database friendly which will make it easier to complete this assignment.


### Store the datasets in database tables

To analyze the data using SQL, it first needs to be loaded into SQLite DB.
We will create three tables in as under:

1.  **CENSUS_DATA**
2.  **CHICAGO_PUBLIC_SCHOOLS**
3.  **CHICAGO_CRIME_DATA**

Let us now load the ipython-sql  extension and establish a connection with the database

* Here you will be loading the csv files into the pandas Dataframe and then loading the data into the above mentioned sqlite tables.

* Next you will be connecting to the sqlite database  **FinalDB**.

Refer to the previous lab for hints .

<a href ="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork/labs/Module%205/DB0201EN-Week3-1-4-Analyzing_SQLite.ipynb">Hands-on Lab: Analyzing a real World Data Set</a>





In [1]:
%load_ext sql

In [2]:
import csv, sqlite3

connection = sqlite3.connect("FinalDB.db")
cursor = connection.cursor()

%sql sqlite:///FinalDB.db

'Connected: @FinalDB.db'

In [3]:
!pip install -q pandas==1.1.5

In [4]:
import pandas as pd

census_data = pd.read_csv("Census_Data.csv")
census_data.to_sql("CENSUS_DATA", connection,  if_exists="replace", index=False, method="multi")

crime_data = pd.read_csv("Chicago_Crime_Data.csv")
crime_data.to_sql("CHICAGO_CRIME_DATA", connection, if_exists="replace", index=False, method="multi")

public_schools_data = pd.read_csv("Chicago_Public_Schools.csv")
public_schools_data.to_sql("CHICAGO_PUBLIC_SCHOOLS", connection, if_exists="replace", index=False, method="multi")

/home/jupyterlab/conda/envs/python/lib/python3.7/site-packages/pandas/core/generic.py:2615: UserWarning: The spaces in these column names will not be changed. In pandas versions < 0.14, spaces were converted to underscores.
  method=method,


In [5]:
%%sql

SELECT * 
FROM CHICAGO_PUBLIC_SCHOOLS
LIMIT 3;

 * sqlite:///FinalDB.db
Done.


[(610038, 'Abraham Lincoln Elementary School', 'ES', '615 W Kemper Pl', 'Chicago', 'IL', 60614, '(773) 534-5720', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610038.pdf', 'Fullerton Elementary Network', 'NORTH-NORTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 1', 'Yes', 'Very Strong', 99.0, 'Very Strong', '99', 'Strong', 74.0, 'Strong', 66.0, 'Weak', '65', 'Strong', '70', 'Strong', '56', 'Average', '47', '96.00%', 2.0, '96.40%', '95.80%', '80.1', '43.3', '89.6', '84.9', '60.7', '62.6', '81.9', '85.2', '52', '62.4', '66.3', '77.9', 69.7, 64.4, 0.2, 0.9, 'Yellow', 'Green', '67.1', '54.5', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 813, 33, 'NDA', 1171699.4579999999, 1915829.428, 41.92449696, -87.64452163, 7, 'LINCOLN PARK', 43, 18, '(41.92449696, -87.64452163)'),
 (610281, 'Adam Clayton Powell Paideia Community Academy Elementary School', 'ES', '7511 S South Shore Dr', 'Chicago', 'IL', 60649, '(773) 535-6650', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610281.pdf', 'Skyway Elementary Network', 'SOUTH SIDE COLLABORATIVE', 'No', 'Track_E', 'Not on Probation', 'Level 1', 'No', 'Average', 54.0, 'Strong', '66', 'Strong', 74.0, 'Very Strong', 84.0, 'Weak', '63', 'Strong', '76', 'Weak', '46', 'Average', '50', '95.60%', 15.7, '95.30%', '100.00%', '62.4', '51.7', '21.9', '15.1', '29', '42.8', '38.5', '27.4', '44.8', '42.7', '14.1', '34.4', 16.8, 16.5, 0.7, 1.4, 'Green', 'Green', '17.2', '27.3', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 521, 46, 'NDA', 1196129.985, 1856209.466, 41.76032435, -87.55673627, 43, 'SOUTH SHORE', 7, 4, '(41.76032435, -87.55673627)'),
 (610185, 'Adlai E Stevenson Elementary School', 'ES', '8010 S Kostner Ave', 'Chicago', 'IL', 60652, '(773) 535-2280', 'http://schoolreports.cps.edu/SchoolProgressReport_Eng/Spring2011Eng_610185.pdf', 'Midway Elementary Network', 'SOUTHWEST SIDE COLLABORATIVE', 'No', 'Standard', 'Not on Probation', 'Level 2', 'No', 'Strong', 61.0, 'NDA', 'NDA', 'Average', 50.0, 'Weak', 36.0, 'Weak', 'NDA', 'NDA', 'NDA', 'Average', '47', 'Weak', '41', '95.70%', 2.3, '94.70%', '98.30%', '53.7', '26.6', '38.3', '34.7', '43.7', '57.3', '48.8', '39.2', '46.8', '44', '7.5', '21.9', 18.3, 15.5, -0.9, -1.0, 'Red', 'Red', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 'NDA', 1324, 44, 'NDA', 1148427.165, 1851012.215, 41.74711093, -87.73170248, 70, 'ASHBURN', 13, 8, '(41.74711093, -87.73170248)')]

In [6]:
%%sql

SELECT * 
FROM CHICAGO_CRIME_DATA
LIMIT 3;

 * sqlite:///FinalDB.db
Done.


[(3512276, 'HK587712', '2004-08-28', '047XX S KEDZIE AVE', '890', 'THEFT', 'FROM BUILDING', 'SMALL RETAIL STORE', 0, 0, 911, 9, 14.0, 58.0, '6', 1155838.0, 1873050.0, 2004, 41.807440500000006, -87.70395585, '(41.8074405, -87.703955849)'),
 (3406613, 'HK456306', '2004-06-26', '009XX N CENTRAL PARK AVE', '820', 'THEFT', '$500 AND UNDER', 'OTHER', 0, 0, 1112, 11, 27.0, 23.0, '6', 1152206.0, 1906127.0, 2004, 41.89827996, -87.71640551, '(41.898279962, -87.716405505)'),
 (8002131, 'HT233595', '2011-04-04', '043XX S WABASH AVE', '820', 'THEFT', '$500 AND UNDER', 'NURSING HOME/RETIREMENT HOME', 0, 0, 221, 2, 3.0, 38.0, '6', 1177436.0, 1876313.0, 2011, 41.81593313, -87.62464213, '(41.815933131, -87.624642127)')]

In [7]:
%%sql

SELECT * 
FROM CENSUS_DATA
LIMIT 3;

 * sqlite:///FinalDB.db
Done.


[(1.0, 'Rogers Park', 7.7, 23.6, 8.7, 18.2, 27.5, 23939, 39.0),
 (2.0, 'West Ridge', 7.8, 17.2, 8.8, 20.8, 38.5, 23040, 46.0),
 (3.0, 'Uptown', 3.8, 24.0, 8.9, 11.8, 22.2, 35787, 20.0)]

In [8]:
%sql SELECT name,type,length(type) FROM PRAGMA_TABLE_INFO('CENSUS_DATA');

 * sqlite:///FinalDB.db
Done.


[('COMMUNITY_AREA_NUMBER', 'REAL', 4),
 ('COMMUNITY_AREA_NAME', 'TEXT', 4),
 ('PERCENT_OF_HOUSING_CROWDED', 'REAL', 4),
 ('PERCENT_HOUSEHOLDS_BELOW_POVERTY', 'REAL', 4),
 ('PERCENT_AGED_16__UNEMPLOYED', 'REAL', 4),
 ('PERCENT_AGED_25__WITHOUT_HIGH_SCHOOL_DIPLOMA', 'REAL', 4),
 ('PERCENT_AGED_UNDER_18_OR_OVER_64', 'REAL', 4),
 ('PER_CAPITA_INCOME', 'INTEGER', 7),
 ('HARDSHIP_INDEX', 'REAL', 4)]

In [9]:
%sql SELECT name,type,length(type) FROM PRAGMA_TABLE_INFO('CHICAGO_CRIME_DATA');

 * sqlite:///FinalDB.db
Done.


[('ID', 'INTEGER', 7),
 ('CASE_NUMBER', 'TEXT', 4),
 ('DATE', 'TEXT', 4),
 ('BLOCK', 'TEXT', 4),
 ('IUCR', 'TEXT', 4),
 ('PRIMARY_TYPE', 'TEXT', 4),
 ('DESCRIPTION', 'TEXT', 4),
 ('LOCATION_DESCRIPTION', 'TEXT', 4),
 ('ARREST', 'INTEGER', 7),
 ('DOMESTIC', 'INTEGER', 7),
 ('BEAT', 'INTEGER', 7),
 ('DISTRICT', 'INTEGER', 7),
 ('WARD', 'REAL', 4),
 ('COMMUNITY_AREA_NUMBER', 'REAL', 4),
 ('FBICODE', 'TEXT', 4),
 ('X_COORDINATE', 'REAL', 4),
 ('Y_COORDINATE', 'REAL', 4),
 ('YEAR', 'INTEGER', 7),
 ('LATITUDE', 'REAL', 4),
 ('LONGITUDE', 'REAL', 4),
 ('LOCATION', 'TEXT', 4)]

In [10]:
%sql SELECT name,type,length(type) FROM PRAGMA_TABLE_INFO('CHICAGO_PUBLIC_SCHOOLS');

 * sqlite:///FinalDB.db
Done.


[('School_ID', 'INTEGER', 7),
 ('NAME_OF_SCHOOL', 'TEXT', 4),
 ('Elementary, Middle, or High School', 'TEXT', 4),
 ('Street_Address', 'TEXT', 4),
 ('City', 'TEXT', 4),
 ('State', 'TEXT', 4),
 ('ZIP_Code', 'INTEGER', 7),
 ('Phone_Number', 'TEXT', 4),
 ('Link', 'TEXT', 4),
 ('Network_Manager', 'TEXT', 4),
 ('Collaborative_Name', 'TEXT', 4),
 ('Adequate_Yearly_Progress_Made_', 'TEXT', 4),
 ('Track_Schedule', 'TEXT', 4),
 ('CPS_Performance_Policy_Status', 'TEXT', 4),
 ('CPS_Performance_Policy_Level', 'TEXT', 4),
 ('HEALTHY_SCHOOL_CERTIFIED', 'TEXT', 4),
 ('Safety_Icon', 'TEXT', 4),
 ('SAFETY_SCORE', 'REAL', 4),
 ('Family_Involvement_Icon', 'TEXT', 4),
 ('Family_Involvement_Score', 'TEXT', 4),
 ('Environment_Icon', 'TEXT', 4),
 ('Environment_Score', 'REAL', 4),
 ('Instruction_Icon', 'TEXT', 4),
 ('Instruction_Score', 'REAL', 4),
 ('Leaders_Icon', 'TEXT', 4),
 ('Leaders_Score', 'TEXT', 4),
 ('Teachers_Icon', 'TEXT', 4),
 ('Teachers_Score', 'TEXT', 4),
 ('Parent_Engagement_Icon', 'TEXT', 4),
 ('Parent_Engagement_Score', 'TEXT', 4),
 ('Parent_Environment_Icon', 'TEXT', 4),
 ('Parent_Environment_Score', 'TEXT', 4),
 ('AVERAGE_STUDENT_ATTENDANCE', 'TEXT', 4),
 ('Rate_of_Misconducts__per_100_students_', 'REAL', 4),
 ('Average_Teacher_Attendance', 'TEXT', 4),
 ('Individualized_Education_Program_Compliance_Rate', 'TEXT', 4),
 ('Pk_2_Literacy__', 'TEXT', 4),
 ('Pk_2_Math__', 'TEXT', 4),
 ('Gr3_5_Grade_Level_Math__', 'TEXT', 4),
 ('Gr3_5_Grade_Level_Read__', 'TEXT', 4),
 ('Gr3_5_Keep_Pace_Read__', 'TEXT', 4),
 ('Gr3_5_Keep_Pace_Math__', 'TEXT', 4),
 ('Gr6_8_Grade_Level_Math__', 'TEXT', 4),
 ('Gr6_8_Grade_Level_Read__', 'TEXT', 4),
 ('Gr6_8_Keep_Pace_Math_', 'TEXT', 4),
 ('Gr6_8_Keep_Pace_Read__', 'TEXT', 4),
 ('Gr_8_Explore_Math__', 'TEXT', 4),
 ('Gr_8_Explore_Read__', 'TEXT', 4),
 ('ISAT_Exceeding_Math__', 'REAL', 4),
 ('ISAT_Exceeding_Reading__', 'REAL', 4),
 ('ISAT_Value_Add_Math', 'REAL', 4),
 ('ISAT_Value_Add_Read', 'REAL', 4),
 ('ISAT_Value_Add_Color_Math', 'TEXT', 4),
 ('ISAT_Value_Add_Color_Read', 'TEXT', 4),
 ('Students_Taking__Algebra__', 'TEXT', 4),
 ('Students_Passing__Algebra__', 'TEXT', 4),
 ('9th Grade EXPLORE (2009)', 'TEXT', 4),
 ('9th Grade EXPLORE (2010)', 'TEXT', 4),
 ('10th Grade PLAN (2009)', 'TEXT', 4),
 ('10th Grade PLAN (2010)', 'TEXT', 4),
 ('Net_Change_EXPLORE_and_PLAN', 'TEXT', 4),
 ('11th Grade Average ACT (2011)', 'TEXT', 4),
 ('Net_Change_PLAN_and_ACT', 'TEXT', 4),
 ('College_Eligibility__', 'TEXT', 4),
 ('Graduation_Rate__', 'TEXT', 4),
 ('College_Enrollment_Rate__', 'TEXT', 4),
 ('COLLEGE_ENROLLMENT', 'INTEGER', 7),
 ('General_Services_Route', 'INTEGER', 7),
 ('Freshman_on_Track_Rate__', 'TEXT', 4),
 ('X_COORDINATE', 'REAL', 4),
 ('Y_COORDINATE', 'REAL', 4),
 ('Latitude', 'REAL', 4),
 ('Longitude', 'REAL', 4),
 ('COMMUNITY_AREA_NUMBER', 'INTEGER', 7),
 ('COMMUNITY_AREA_NAME', 'TEXT', 4),
 ('Ward', 'INTEGER', 7),
 ('Police_District', 'INTEGER', 7),
 ('Location', 'TEXT', 4)]

## Problems

Now write and execute SQL queries to solve assignment problems

### Problem 1

##### Find the total number of crimes recorded in the CRIME table.


In [11]:
%%sql
SELECT COUNT(*)
FROM CHICAGO_CRIME_DATA;

 * sqlite:///FinalDB.db
Done.


[(533,)]

### Problem 2

##### List community areas with per capita income less than 11000.


In [12]:
%%sql
SELECT COMMUNITY_AREA_NAME, PER_CAPITA_INCOME
FROM CENSUS_DATA
WHERE PER_CAPITA_INCOME < 11000;

 * sqlite:///FinalDB.db
Done.


[('West Garfield Park', 10934),
 ('South Lawndale', 10402),
 ('Fuller Park', 10432),
 ('Riverdale', 8201)]

### Problem 3

##### List all case numbers for crimes  involving minors?(children are not considered minors for the purposes of crime analysis)


In [13]:
%%sql 
SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION
FROM CHICAGO_CRIME_DATA
WHERE PRIMARY_TYPE LIKE '%MINOR%' OR DESCRIPTION LIKE '%MINOR%';

 * sqlite:///FinalDB.db
Done.


[('HL266884', 'LIQUOR LAW VIOLATION', 'SELL/GIVE/DEL LIQUOR TO MINOR'),
 ('HK238408', 'LIQUOR LAW VIOLATION', 'ILLEGAL CONSUMPTION BY MINOR')]

### Problem 4

##### List all kidnapping crimes involving a child?


In [14]:
%%sql 
SELECT CASE_NUMBER, PRIMARY_TYPE, DESCRIPTION
FROM CHICAGO_CRIME_DATA
WHERE PRIMARY_TYPE LIKE '%KIDNAP%' AND DESCRIPTION LIKE '%CHILD%';

 * sqlite:///FinalDB.db
Done.


[('HN144152', 'KIDNAPPING', 'CHILD ABDUCTION/STRANGER')]

### Problem 5

##### What kinds of crimes were recorded at schools?


In [15]:
%%sql 
SELECT PRIMARY_TYPE, DESCRIPTION
FROM CHICAGO_CRIME_DATA
WHERE LOCATION_DESCRIPTION LIKE '%SCHOOL%'
GROUP BY (PRIMARY_TYPE);

 * sqlite:///FinalDB.db
Done.


[('ASSAULT', 'PRO EMP HANDS NO/MIN INJURY'),
 ('BATTERY', 'SIMPLE'),
 ('CRIMINAL DAMAGE', 'TO VEHICLE'),
 ('CRIMINAL TRESPASS', 'TO LAND'),
 ('NARCOTICS', 'POSS: HEROIN(WHITE)'),
 ('PUBLIC PEACE VIOLATION', 'BOMB THREAT')]

### Problem 6

##### List the average safety score for each type of school.


In [16]:
%%sql 
SELECT "Elementary, Middle, or High School" AS EMH,  AVG(SAFETY_SCORE) AS AVERAGE_SAFETY_SCORE
FROM CHICAGO_PUBLIC_SCHOOLS
GROUP BY EMH;

 * sqlite:///FinalDB.db
Done.


[('ES', 49.52038369304557), ('HS', 49.62352941176471), ('MS', 48.0)]

### Problem 7

##### List 5 community areas with highest % of households below poverty line


In [17]:
%%sql
SELECT COMMUNITY_AREA_NAME, PERCENT_HOUSEHOLDS_BELOW_POVERTY
FROM CENSUS_DATA
ORDER BY PERCENT_HOUSEHOLDS_BELOW_POVERTY DESC 
LIMIT 5;

 * sqlite:///FinalDB.db
Done.


[('Riverdale', 56.5),
 ('Fuller Park', 51.2),
 ('Englewood', 46.6),
 ('North Lawndale', 43.1),
 ('East Garfield Park', 42.4)]

### Problem 8

##### Which community area is most crime prone?


In [18]:
%%sql
SELECT CD.COMMUNITY_AREA_NAME, CC.COMMUNITY_AREA_NUMBER, COUNT(*) AS CRIMES_PER_COMMUNITY
FROM CHICAGO_CRIME_DATA CC, CENSUS_DATA CD
WHERE CC.COMMUNITY_AREA_NUMBER = CD.COMMUNITY_AREA_NUMBER
GROUP BY CC.COMMUNITY_AREA_NUMBER
ORDER BY CRIMES_PER_COMMUNITY DESC 
LIMIT 1;

 * sqlite:///FinalDB.db
Done.


[('Austin', 25.0, 43)]

Double-click **here** for a hint

<!--
Query for the 'community area number' that is most crime prone.
-->


### Problem 9

##### Use a sub-query to find the name of the community area with highest hardship index


In [20]:
%%sql 
SELECT COMMUNITY_AREA_NAME, HARDSHIP_INDEX
FROM CENSUS_DATA
WHERE HARDSHIP_INDEX = (SELECT MAX(HARDSHIP_INDEX) 
                        FROM CENSUS_DATA);

 * sqlite:///FinalDB.db
Done.


[('Riverdale', 98.0)]

### Problem 10

##### Use a sub-query to determine the Community Area Name with most number of crimes?


In [21]:
%%sql 
SELECT COMMUNITY_AREA_NAME
FROM CENSUS_DATA
WHERE COMMUNITY_AREA_NUMBER = (SELECT COMMUNITY_AREA_NUMBER
                                FROM CHICAGO_CRIME_DATA
                                WHERE COMMUNITY_AREA_NUMBER != 'None'
                                GROUP BY COMMUNITY_AREA_NUMBER
                                ORDER BY COUNT(COMMUNITY_AREA_NUMBER)DESC
                                LIMIT 1);
                               

 * sqlite:///FinalDB.db
Done.


[('Austin',)]

Copyright © 2020 This notebook and its source code are released under the terms of the [MIT License](https://bigdatauniversity.com/mit-license?utm_medium=Exinfluencer&utm_source=Exinfluencer&utm_content=000026UJ&utm_term=10006555&utm_id=NA-SkillsNetwork-Channel-SkillsNetworkCoursesIBMDeveloperSkillsNetworkDB0201ENSkillsNetwork20127838-2021-01-01&cm_mmc=Email_Newsletter-_-Developer_Ed%2BTech-_-WW_WW-_-SkillsNetwork-Courses-IBMDeveloperSkillsNetwork-DB0201EN-SkillsNetwork-20127838&cm_mmca1=000026UJ&cm_mmca2=10006555&cm_mmca3=M12345678&cvosrc=email.Newsletter.M12345678&cvo_campaign=000026UJ).


## Author(s)

<h4> Hima Vasudevan </h4>
<h4> Rav Ahuja </h4>
<h4> Ramesh Sannreddy </h4>

## Contribtuor(s)

<h4> Malika Singla </h4>

## Change log

| Date       | Version | Changed by        | Change Description                             |
| ---------- | ------- | ----------------- | ---------------------------------------------- |
| 2022-03-04 | 2.5     | Lakshmi Holla     | Changed markdown.                   |
| 2021-05-19 | 2.4     | Lakshmi Holla     | Updated the question                           |
| 2021-04-30 | 2.3     | Malika Singla     | Updated the libraries                          |
| 2021-01-15 | 2.2     | Rav Ahuja         | Removed problem 11 and fixed changelog         |
| 2020-11-25 | 2.1     | Ramesh Sannareddy | Updated the problem statements, and datasets   |
| 2020-09-05 | 2.0     | Malika Singla     | Moved lab to course repo in GitLab             |
| 2018-07-18 | 1.0     | Rav Ahuja         | Several updates including loading instructions |
| 2018-05-04 | 0.1     | Hima Vasudevan    | Created initial version                        |

## <h3 align="center"> © IBM Corporation 2020. All rights reserved. <h3/>
